lib/Elevate/Components/CCS.pm

```diff
@@ -0,0 +1,424 @@
+use constant TASK_QUEUE_SCRIPT  => '/usr/local/cpanel/bin/servers_queue';
+
+use constant DUMP_TYPES => (
+    calendars => 'ics',
+    contacts  => 'vcard',
+);
+
+sub pre_leapp ($self) {
+    my $ccs_installed = Cpanel::Pkgr::is_installed(CCS_PACKAGE);
+    Elevate::StageFile::update_stage_file( { ccs_installed => $ccs_installed } );
+    return unless $ccs_installed;
+
+    $self->_load_ccs_modules();
+
+    $self->run_once('export_ccs_data');
+    $self->remove_ccs_and_dependencies();
+
+    # Removing the PKG will leave this directory in place
+    # This results in PostGreSQL/CCS failing to start after leapp completes
+    $self->clean_up_pkg_cruft();
+
+    return;
+}
+
+sub clean_up_pkg_cruft ($self) {
+    $self->move_pgsql_directory();
+    $self->remove_cpanel_ccs_home_directory();
+    return;
+}
+
+=head1 remove_cpanel_ccs_home_directory
+
+Removing the package removes the `cpanel-ccs` user, but leaves behind
+the data in the home directory.  This results in the service failing
+to start after the elevation has completed
+
+=cut
+
+sub remove_cpanel_ccs_home_directory ($self) {
+    File::Path::remove_tree('/opt/cpanel-ccs') if -d '/opt/cpanel-ccs';
+    return;
+}
+
+=head1 move_pgsql_directory
+
+Removing the PKG will leave this directory in place
+This results in PostGreSQL/CCS failing to start after leapp completes
+
+=cut
+
+sub move_pgsql_directory ($self) {
+    my $pg_dir        = '/var/lib/pgsql';
+    my $pg_backup_dir = '/var/lib/pgsql_pre_elevate';
+
+    # Remove the backup path if it exists as a directory
+    File::Path::remove_tree($pg_backup_dir) if -e $pg_backup_dir && -d $pg_backup_dir;
+
+    # If we were unable to remove the backup path above, then change it to something that
+    # should be unique
+    $pg_backup_dir .= '_' . time() . '_' . $$ if -e $pg_backup_dir;
+
+    # Make sure the path that should be unique does not exist
+    File::Path::remove_tree($pg_backup_dir) if -e $pg_backup_dir && -d $pg_backup_dir;
+
+    # Give it up if we still do not have a candidate to back the data up to
+    if ( -e $pg_backup_dir ) {
+        die <<~"EOS";
+        Unable to ensure a valid backup path for $pg_dir.
+        Please ensure that '/var/lib/pgsql_pre_elevate' does not exist on your system and execute this script again with
+
+        /scripts/elevate-cpanel --continue
+
+        EOS
+    }
+
+    INFO( <<~"EOS" );
+    Moving the PostgreSQL data dir located at $pg_dir to $pg_backup_dir
+    to ensure a functioning PostgreSQL server after the elevation completes.
+    EOS
+
+    File::Copy::move( $pg_dir, $pg_backup_dir ) if -d $pg_dir;
+
+    return;
+}
+
+sub remove_ccs_and_dependencies ($self) {
+
+    # There are other dependencies but these are the 3 that we are concerned with
+    my @ccs_depedencies = qw{
+      postgresql
+      postgresql-devel
+      postgresql-server
+    };
+
+    $self->yum->remove( CCS_PACKAGE(), @ccs_depedencies );
```

Comment:
I'm ok with this but I am worried what we'll ALSO remove that we weren't expecting to.

Reply:
We do block on customer owned PG DBs.  The only way we get this far is if there is no customer owned PG data.